src/router/index.js

```diff
@@ -107,6 +111,22 @@ const _mutationHandlers = {
   }
 }
 
+const checkInitialQuery = (router, state) => {
+  let initialQuery = state.initialQuery
+  for (let i = 0; i < state.linkSchemes.length; i++) {
+    let scheme = state.linkSchemes[i]
```

Comment:
I used a plain loop because of
> There is no way to stop or break a forEach() loop other than by throwing an exception. If you need such behavior, the forEach() method is the wrong tool. Use a plain loop instead.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

to stop after the first matching route has been found. I think even using return can lead to unexpected behaviour

Reply:
Ok, I understood. Let's use `Array.find` instead, so this code can look like:
```
const checkInitialQuery = (router, { initialQuery, linkSchemes }) =>
  !!linkSchemes.find(scheme => typeof scheme === 'function' && scheme(initialQuery))
```